src/main/java/org/opensearch/knn/index/query/KNNQueryFactory.java

```diff
@@ -53,6 +55,7 @@ public static Query create(
         String fieldName,
         float[] vector,
         int k,
+        Float radius,
```

Comment:
On second thought, we should have a separate factory that is called for Radial Queries. Putting k and radius in same create method is confusing. There should be a RadialQueryFactory that gets called if we need a radial nearest neighbor query and KNNQueryFactory gets called if we need a KNNQuery. @VijayanB what do you think?

Reply:
Since all current files and class name are originally from KNN, I'm not sure if we want a more generic name and pattern from "KNN" to "VectorQuery" in whole codebase, but this is a long term refactoring I think. For now I prefer reuse the existing factory without much change.

Reply:
@jmazanec15 Good point. I like your suggestion

Reply:
@jmazanec15 @VijayanB Added BaseQueryFactory as parent class and added new RNNQueryFactory for radials search.